modyn/protos/evaluator.proto

```diff
@@ -43,36 +47,42 @@ message EvaluateModelRequest {
 }
 
 message EvaluateModelResponse {
+  // only when all interval evaluations failed, this field will be set to false
   bool evaluation_started = 1;
   int32 evaluation_id = 2;
-  int64 dataset_size = 3;
-  EvaluationAbortedReason eval_aborted_reason = 4;
+  // non-empty only when evaluation_started is true; in this case it has the same size as the number of intervals
+  repeated int64 dataset_sizes = 3;
+  // always has the same size as the number of intervals
+  repeated EvaluationAbortedReason eval_aborted_reasons = 4;
```

Comment:
If I understood correctly, an evaluation for a certain interval can fail because of 
1) a problem with the interval (e.g. no data in that interval)
2) a problem with whole request (applies to all intervals in the request)

If that is the case I think the cleanest solution would be to have an error response field on request level and an error field on interval level.

That goes in hand with the `repeated int64 dataset_sizes`. Currently, we can only match the `data_sizes` to the error codes and intervals by assuming consistent indexes in the sequences.

I recommend to add a `repeated EvaluateModelIntervalResponse` to the `EvaluateModelResponse` and add all interval-specific things (dataset_sizes, interval failure, ...) there.

Reply:
make sense. will do

Reply:
added; make sense thanks